### DISCUSSION TITLE
mip: install file on host

### DISCUSSION THREAD

**Discussion by iot49** [micropython/micropython#14261] [General]
How can I use mip (or mpremote) to install a library on the host, e.g. for freezing?

From the documentation it seems that mip installs libraries only in a micropython VM, even when run on the host via mpremote.

**Comment by andrewleech** [comment]
For compiling micropython with frozen code you use a cousin manifest.json file with `require()` entries and similar for the libraries you would mip install onto pre-built firmware.

**Reply by iot49** [reply]
Yes, but won’t mip then install on the microcontroller? Also, that’s where it is installed and runs. Currently, as a workaround, I use mip in the unix port of micropython. I hoped for a simpler solution that can be used easily with github actions.

**Reply by andrewleech** [reply]
I'm not sure what you mean about "won't mip then install on the microcontroller?"
Once a library is frozen into code mip isn't needed.

**Reply by andrewleech** [reply]
Or do you mean you want to run code on the unix port that needs some mip installable libraries? For that you should be able to just do `micropython -m mip install <the library>`
